src/org.ts

```diff
@@ -211,6 +211,49 @@ export class Org extends AsyncCreatable<Org.Options> {
     }
   }
 
+  /**
+   * If this Org is a scratch org, calling this method will delete the scratch org from the DevHub and clean up any local files
+   *
+   * @param hubOrg - optional DevHub Org of the to-be-deleted scratch org
+   */
+  public async deleteScratchOrg(hubOrg?: Org): Promise<void> {
+    const logger = await Logger.child('deleteScratchOrg');
+
+    // if we didn't get a hubOrg, we'll get it from the this org
+    hubOrg = hubOrg ?? (await this.getDevHubOrg());
+    if (hubOrg?.getOrgId() === this.getOrgId()) {
+      // we're attempting to delete a DevHub
+      throw SfdxError.create('@salesforce/core', 'org', 'deleteOrgHubError');
+    }
+    if (hubOrg) {
```

Comment:
what's the scenario where there's not a hubOrg here?  Trying to delete something that doesn't have a hub (like another non-scratch org?)

Would it be better to throw something instead of returning?